packages/block-library/src/embed/embed-bottom-sheet.native.js

```diff
@@ -11,28 +11,34 @@ import { useDispatch } from '@wordpress/data';
 import { store as noticesStore } from '@wordpress/notices';
 import { useCallback, useState } from '@wordpress/element';
 
-const linkSettingsOptions = {
-	url: {
-		label: __( 'Embed link' ),
-		placeholder: __( 'Add link' ),
-		autoFocus: true,
-		autoFill: true,
-	},
-	footer: {
-		label: (
-			<FooterMessageLink
-				href={ __( 'https://wordpress.org/support/article/embeds/' ) }
-				value={ __( 'Learn more about embeds' ) }
-			/>
-		),
-		separatorType: 'topFullWidth',
-	},
-};
-
-const EmbedBottomSheet = ( { value, isVisible, onClose, onSubmit } ) => {
+const EmbedBottomSheet = ( { value, label, isVisible, onClose, onSubmit } ) => {
 	const [ url, setURL ] = useState( value );
 	const { createErrorNotice } = useDispatch( noticesStore );
 
+	const linkSettingsOptions = {
+		url: {
+			label: sprintf(
+				// translators: %s: embed block variant's label e.g: "Twitter".
+				__( '%s link' ),
+				label
+			),
+			placeholder: __( 'Add link' ),
+			autoFocus: true,
+			autoFill: true,
+		},
+		footer: {
+			label: (
+				<FooterMessageLink
+					href={ __(
+						'https://wordpress.org/support/article/embeds/'
+					) }
+					value={ __( 'Learn more about embeds' ) }
+				/>
+			),
+			separatorType: 'topFullWidth',
+		},
+	};
```

Comment:
I'm wondering if we could memoize this with `useMemo` hook, I think it will help us preventing potential re-renderings as the component creates a new object on every pass.

Reply:
I tried adding `useMemo` here and looks like it doesn't reduce render calls, I don't have a strong feeling regarding whether to add it or not so I let you decide on this 😅 .

Reply:
I think this component won't be on screen most of the time and re-renders shouldn't be too expensive, so I'll skip it for now, until we perceive any performance issues with this 😅